addOns/quickstart/src/main/java/org/zaproxy/zap/extension/quickstart/AttackThread.java

```diff
@@ -89,11 +99,22 @@ public void run() {
                 // the problem
                 return;
             }
+
+            // what the heck am I supposed to do here?
             if (stopAttack) {
                 LOGGER.debug("Attack stopped manually");
                 extension.notifyProgress(Progress.stopped);
                 return;
             }
+
+            if (pauseAttack) {
+                LOGGER.debug("Attack paused manually");
+                // extension.notifyProgress(Progress.paused);
+                return;
```

Comment:
These pause checks outside the "wait loops" should wait instead of return (which would behave like stopping).